controllers/external-accounts.js

```diff
@@ -71,14 +71,20 @@ const linkExternalAccount = async (req, res) => {
       userId
     );
 
-    try {
-      const unverifiedRoleId = config.get("discordUnverifiedRoleId");
-      await discordServices.removeRoleFromUser(unverifiedRoleId, attributes.discordId, req.userData);
-    } catch (error) {
-      logger.error(`Error getting external account data: ${error}`);
-      return res.boom.internal("Role Deletion failed. Please contact admin.", {
-        message: "Role Deletion failed. Please contact admin.",
-      });
+    const unverifiedRoleId = config.get("discordUnverifiedRoleId");
+    const unverifiedRoleRemovalResponse = await removeDiscordRoleUtils.removeDiscordRoleFromUser(
+      req.userData,
+      attributes.discordId,
+      unverifiedRoleId
+    );
+
+    if (!unverifiedRoleRemovalResponse.success) {
+      return res.boom.internal(
+        `User details updated but ${unverifiedRoleRemovalResponse.message}. Please contact admin`,
```

Comment:
This messages is repeated twice.

Reply:
boom.internal is not throwing the custom message if I don't user it twice like this. It is throwing its default message in that case. That's why I'm sending "message" property in the body with that message so that if it fails to take the first one it will definitely take the second one
![image](https://github.com/user-attachments/assets/1dd3a2d3-41a6-45b6-99f6-5e9a0238c990)
![image](https://github.com/user-attachments/assets/03f0e0bf-ff7a-4cc5-b822-df2d8f1b6fa3)

Reply:
There is another method for 500. Have you checked?

Reply:
Are you asking about using normal res.status(500) ? If that, I did use that in my previous task. Prakash asked me to use boom

Reply:
res.boom.serverUnavailable

Reply:
Please have a look at this https://github.com/Real-Dev-Squad/website-backend/pull/2236#discussion_r1826583987

Reply:
What is the first and second argument of internal function then?

Reply:
message and data
This is how it is coming in this case
![image](https://github.com/user-attachments/assets/d461de0f-1bc0-4945-b412-60ce511050ef)

Reply:
```unverifiedRoleRemovalResponse.message```
what will be the value of this?

Reply:
For failure cases - Role deleion failed or Role doesn't exist
For success case - Role deleted successfully.

The function already has these. You can see there

Reply:
In line 23 of utils function can throw any error which will be different from these two case. So are we going to return that error to client?

Reply:
I've updated the code. Can you please check it?